NitroxPatcher/Patches/Dynamic/SpawnOnKill_OnKill_Patch.cs

```diff
@@ -0,0 +1,59 @@
+using HarmonyLib;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using NitroxPatcher.PatternMatching;
+using Oculus.Platform;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEngine;
+using static System.Reflection.Emit.OpCodes;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Synchronizes entities that Spawn something when they are killed, e.g. Coral Disks.
+/// </summary>
+public class SpawnOnKill_OnKill_Patch : NitroxPatch, IDynamicPatch
+{
+    public static readonly MethodInfo TARGET_METHOD = Reflect.Method((SpawnOnKill t) => t.OnKill());
+
+    private static readonly InstructionsPattern spawnInstanceOnKillPattern = new()
+    {
+        Reflect.Method(() => Object.Instantiate(default(GameObject), default(Vector3), default(Quaternion))),
+        { Stloc_0, "DropOnKillInstance" },
+        Ldarg_0,
+    };
+
+    public static IEnumerable<CodeInstruction> Transpiler(MethodBase original, IEnumerable<CodeInstruction> instructions)
+    {
+        static IEnumerable<CodeInstruction> InsertCallbackCall(string label, CodeInstruction _)
```

Comment:
I'm not sure why you created a function inside while you could just apply .Transform just like in [DevConsole_Update_Patch](https://github.com/SubnauticaNitrox/Nitrox/blob/master/NitroxPatcher/Patches/Dynamic/DevConsole_Update_Patch.cs)